library/alloc/src/lib.rs

```diff
@@ -130,6 +130,8 @@
 #![feature(extend_one)]
 #![feature(fmt_internals)]
 #![feature(fn_traits)]
+#![feature(freeze)]
+#![feature(freeze_impls)]
```

Comment:
This feature was never meant to be used outside libcore :/ I'm beginning to regret that we made this type public again...

Reply:
What if instead of this custom `Freeze` thing we implement a new builtin type like `AlignAs<T>`, which will have all properties like `Copy`, `Sync` and `Freeze`, but without inheriting properties of `T` except alignment?

That would solve this issue with `Freeze` (but won't solve the padding issue from the thread above).

Reply:
That's a good idea. We don't even necessarily have to do this now, we could say that's our backup plan if `Freeze` ever becomes un-`impl`ementable.

I still haven't understood the padding issue.